### PR TITLE
Generalise 'AllowNewer'-types' names

### PR DESCRIPTION
**Pull request by hvr**
This also adds a not yet used `AllowOlder` newtype

This is a preparatory refactoring propsed in #3466 for supporting `--allow-older`

### CODE REVIEW

**Comment by hvr** [comment]
@dcoutts @23Skidoo all checks passed; are you ok with this refactoring?

File: Cabal/Distribution/Simple/Setup.hs [hunk 1]
```diff
@@ -263,63 +264,87 @@ instance Semigroup GlobalFlags where
...
+-- | 'RelaxDeps' in the context of lower bounds (i.e. for @--allow-older@ flag)
+newtype AllowOlder = AllowOlder { unAllowOlder :: RelaxDeps }
+                   deriving (Eq, Read, Show, Generic)
+
 -- | Dependencies can be relaxed either for all packages in the install plan, or
 -- only for some packages.
-data AllowNewerDep = AllowNewerDep PackageName
-                   | AllowNewerDepScoped PackageName PackageName
+data RelaxedDep = RelaxedDep PackageName
+                   | RelaxedDepScoped PackageName PackageName
```
Comment (dcoutts) [inline Cabal/Distribution/Simple/Setup.hs]: indentation

Comment (hvr) [inline Cabal/Distribution/Simple/Setup.hs]: fixed

**Comment by 23Skidoo** [comment]
@hvr Haven't reviewed all changes in detail, but fine with me in principle.

File: Cabal/Distribution/Simple/Setup.hs [hunk 2]
```diff
@@ -263,63 +264,87 @@ instance Semigroup GlobalFlags where
 -- 'build-depends: array >= 0.3 && < 0.5', are we allowed to relax the upper
 -- bound and choose a version of 'array' that is greater or equal to 0.5? By
 -- default the upper bounds are always strictly honored.
-data AllowNewer =
+data RelaxDeps =
 
   -- | Default: honor the upper bounds in all dependencies, never choose
```
Comment (phadej) [inline Cabal/Distribution/Simple/Setup.hs]: this haddock is outdated

**Comment by ezyang** [comment]
Merged.

**Pull request closed without merging**